receiver/sqlserverreceiver/scraper.go

```diff
@@ -102,6 +121,15 @@ func (s *sqlServerScraperHelper) ScrapeMetrics(ctx context.Context) (pmetric.Met
 	return s.mb.Emit(), nil
 }
 
+func (s *sqlServerScraperHelper) ScrapeLogs(ctx context.Context) (plog.Logs, error) {
+	switch s.sqlQuery {
+	case getSQLServerQueryTextAndPlanQuery(s.instanceName, s.maxQuerySampleCount, s.lookbackTime):
```

Comment:
are you going to compute that query for equality every time logs are scraped? Any chance this can be done earlier?

Reply:
this seems a change which need to involve some existing queries in this receiver as well. I'd like to create a separate PR for this optimization. @atoulme does that sound good to you?

Reply:
yep np